pmd-java/src/main/java/net/sourceforge/pmd/lang/java/metrics/impl/WocMetric.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.java.metrics.impl;
+
+import net.sourceforge.pmd.lang.java.ast.ASTAnyTypeDeclaration;
+import net.sourceforge.pmd.lang.java.ast.ASTAnyTypeDeclaration.TypeKind;
+import net.sourceforge.pmd.lang.java.metrics.signature.JavaOperationSigMask;
+import net.sourceforge.pmd.lang.java.metrics.signature.JavaOperationSignature.Role;
+import net.sourceforge.pmd.lang.java.metrics.signature.JavaSignature.Visibility;
+import net.sourceforge.pmd.lang.metrics.MetricOptions;
+
+/**
+ * Weight of class.
+ *
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+public class WocMetric extends AbstractJavaClassMetric {
+
+    @Override
+    public boolean supports(ASTAnyTypeDeclaration node) {
+        return node.getTypeKind() == TypeKind.CLASS;
+    }
+
+
+    @Override
+    public double computeFor(ASTAnyTypeDeclaration node, MetricOptions options) {
+
+        JavaOperationSigMask mask = new JavaOperationSigMask();
+        mask.restrictRolesTo(Role.METHOD, Role.STATIC);
+        mask.restrictVisibilitiesTo(Visibility.PUBLIC);
+
+        int functionalMethods = countMatchingOpSigs(node, mask);
+
+        mask.coverAllRoles();
+
+        int totalPublicMethods = countMatchingOpSigs(node, mask);
+
+        return functionalMethods / (totalPublicMethods + 1.);
```

Comment:
I would say yes, return NaN - since we can't compute the metric in that case. I think you'll get this now automatically, since the result type is a double.